drivers: crc: nxp build fails on frdm_mcxc444 and mimxrt700_evk/mimxrt798s/cm33_cpu1 after enabling nxp,crc

### Describe the bug

`tests/drivers/build_all` started failing on main after the recent enablement of additional NXP CRC platforms.

Observed failing targets:

- `frdm_mcxc444/mcxc444`
- `mimxrt700_evk/mimxrt798s/cm33_cpu1`

Observed test:

- `drivers.w1.build`

Observed on main at:

- `c2961410b98ff22b771b4803162fa18ac898f9cd`
- `v4.4.0-3391-gc2961410b98f`

- [x] is_regression

454fabb1556bc4d86c83b914d5d4710f5b0c512a is the first bad commit
commit 454fabb1556bc4d86c83b914d5d4710f5b0c512a
Author: Holt Sun <holt.sun@nxp.com>
Date:   Mon May 18 22:00:02 2026 +0800

    drivers: crc: enable NXP CRC platforms
    
    Add nxp,crc devicetree nodes for NXP SoCs that ship the standard
    NXP CRC peripheral, and mark the CRC instance as the Zephyr CRC
    device on the corresponding boards.
    
    This extends the existing NXP CRC driver coverage without changing
    the driver implementation.
    
    Signed-off-by: Holt Sun <holt.sun@nxp.com>

The build failure points into the new NXP CRC driver instantiation path:

```text
.../zephyr/drivers/crc/crc_nxp.c:213:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
213 | DT_INST_FOREACH_STATUS_OKAY(CRC_NXP_INIT)
    | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

This looks like a regression introduced by the recent platform enablement work in `#109331`.

### To Reproduce

From a Zephyr workspace:

```sh
git checkout c2961410b98ff22b771b4803162fa18ac898f9cd
west update
python scripts/twister \
  -p frdm_mcxc444 \
  -p mimxrt700_evk/mimxrt798s/cm33_cpu1 \
  -T tests/drivers/build_all \
  --build-only \
  --inline-logs \
  -M all
```

### Expected behavior

`tests/drivers/build_all` should build successfully on these targets, or the CRC node/driver should not be enabled for targets that do not have usable HAL support.

### Additional context

The likely root cause is that `CONFIG_CRC_DRIVER_NXP` is enabled purely from `DT_HAS_NXP_CRC_ENABLED`, but the underlying NXP HAL does not appear to expose CRC support for these two target/header combinations:

- `modules/hal/nxp/mcux/mcux-sdk-ng/devices/MCX/MCXC/MCXC444/MCXC444_COMMON.h` and `MCXC444_features.h` do not provide CRC symbols / CRC feature macros.
- `modules/hal/nxp/mcux/mcux-sdk-ng/devices/RT/RT700/MIMXRT798S/MIMXRT798S_cm33_core1_COMMON.h` and `_features.h` do not provide CRC symbols, while `cm33_core0` does.

By contrast, supported targets such as `MCXC242` and `MIMXRT798S cm33_core0` do have the expected CRC HAL definitions.

So this seems to be one of:

1. unsupported boards/cores were enabled for `nxp,crc`, or
2. the driver Kconfig needs an additional HAL capability guard, or
3. missing HAL support needs to be added before enabling these targets.

### Impact

This breaks CI for `tests/drivers/build_all` on main for NXP platforms.

--- a/boards/nxp/frdm_mcxc444/frdm_mcxc444.dts
+++ b/boards/nxp/frdm_mcxc444/frdm_mcxc444.dts
@@ -36,7 +36,6 @@
 		zephyr,console = &lpuart0;
 		zephyr,shell-uart = &lpuart0;
 		zephyr,code-partition = &slot0_partition;
-		zephyr,crc = &crc;
 	};
 
 	leds {
@@ -233,8 +232,4 @@ zephyr_udc0: &usb {
 			reg = <0x00038000 DT_SIZE_K(32)>;
 		};
 	};
-};
-
-&crc {
-	status = "okay";
 };

--- a/boards/nxp/mimxrt700_evk/mimxrt700_evk_mimxrt798s_cm33_cpu1.dts
+++ b/boards/nxp/mimxrt700_evk/mimxrt700_evk_mimxrt798s_cm33_cpu1.dts
@@ -28,7 +28,6 @@
 		zephyr,sram = &sram3;
 		zephyr,console = &flexcomm19_lpuart19;
 		zephyr,shell-uart = &flexcomm19_lpuart19;
-		zephyr,crc = &crc;
 	};
 
 	leds {
@@ -139,9 +138,5 @@
 
 &sema420 {
 	domain-id = <5>;
-	status = "okay";
-};
-
-&crc {
 	status = "okay";
 };

--- a/dts/arm/nxp/imxrt/nxp_rt7xx_cm33_cpu0.dtsi
+++ b/dts/arm/nxp/imxrt/nxp_rt7xx_cm33_cpu0.dtsi
@@ -1221,6 +1221,12 @@
 		num-locks = <64>;
 		status = "disabled";
 	};
+
+	crc: crc@151000 {
+		compatible = "nxp,crc";
+		reg = <0x151000 0x1000>;
+		status = "disabled";
+	};
 };
 
 &xspi0 {

--- a/dts/arm/nxp/imxrt/nxp_rt7xx_common.dtsi
+++ b/dts/arm/nxp/imxrt/nxp_rt7xx_common.dtsi
@@ -214,12 +214,6 @@
 		num-locks = <64>;
 		status = "disabled";
 	};
-
-	crc: crc@151000 {
-		compatible = "nxp,crc";
-		reg = <0x151000 0x1000>;
-		status = "disabled";
-	};
 };
 
 &systick {

--- a/dts/arm/nxp/mcx/nxp_mcxc444.dtsi
+++ b/dts/arm/nxp/mcx/nxp_mcxc444.dtsi
@@ -18,3 +18,6 @@
 &flash0 {
 	reg = <0 DT_SIZE_K(256)>;
 };
+
+/* MCXC444 HAL does not expose CRC peripheral support. */
+/delete-node/ &crc;

